controllers/volumereplicationgroup_controller.go

```diff
@@ -592,102 +681,276 @@ func validateS3Endpoint(s3Endpoint, s3Bucket string, log logr.Logger) error {
 	return nil
 }
 
-// createVolumeReplicationCRForPVC creates a VolumeReplication CR with a PVC as its data source.
-func (v *VRGInstance) createVolumeReplicationForPVC(pvc corev1.PersistentVolumeClaim) error {
+// processAsSecondary reconciles the current instance of VRG as secondary
+func (v *VRGInstance) processAsSecondary() (ctrl.Result, error) {
+	requeue := false
+	requeue = v.reconcileVRsAsSecondary()
+
+	// TODO: Update status
+
+	if requeue {
+		v.log.Info("Requeuing resource")
+
+		return ctrl.Result{Requeue: requeue}, nil
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// reconcileVRsAsSecondary reconciles VolumeReplication resources for the VRG as secondary
+func (v *VRGInstance) reconcileVRsAsSecondary() bool {
+	requeue := false
+
+	for idx := range v.pvcList.Items {
+		pvc := &v.pvcList.Items[idx]
+		pvcNamespacedName := types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}
+		log := v.log.WithValues("pvc", pvcNamespacedName.String())
+
+		requeueResult, skip := v.ensurePVCIsSecondary(pvc, log)
+		if requeueResult {
+			requeue = true
+
+			continue
+		}
+
+		if skip {
+			continue
+		}
+
+		if err := v.processVRAsSecondary(pvcNamespacedName, log); err != nil {
+			log.Info("Requeuing due to failure in getting or creating VolumeReplication resource for PersistentVolumeClaim",
+				"errorValue", err)
+
+			requeue = true
+
+			continue
+		}
+
+		log.Info("Successfully processed VolumeReplication for PersistentVolumeClaim")
+	}
+
+	// TODO: We will not unprotect the PVC etc. just check if this is fine, and ok if done during delete
+	// TODO: updateVRStatus even when resource requires a requeue
+
+	return requeue
+}
+
+// ensurePVCIsSecondary ensures a PVC is safe to be marked as Secondary
+func (v *VRGInstance) ensurePVCIsSecondary(pvc *corev1.PersistentVolumeClaim, log logr.Logger) (bool, bool) {
+	const (
+		requeue bool = true
+		skip    bool = true
+	)
+
+	// if the PVC is not bound yet, skip it
+	if pvc.Status.Phase != corev1.ClaimBound {
+		log.Info("Skipping as PersistentVolumeClaim is not bound", "pvcPhase", pvc.Status.Phase)
+
+		return !requeue, skip
+	}
+
+	// Check if PVC is already protected
+	pvcProtected, err := v.ensurePVCVRProtected(pvc, log)
+	if err != nil {
+		log.Info("Requeuing as determining PersistentVolumeClaim protection failed", "errorValue", err)
+
+		return requeue, !skip
+	}
+
+	// Skip if not protected
+	if !pvcProtected {
+		return !requeue, skip
+	}
+
+	// If PVC is not being deleted, requeue
+	if pvc.GetDeletionTimestamp().IsZero() {
+		log.Info("Requeueing as PersistentVolumeClaim is not marked for deletion")
+
+		return requeue, !skip
+	}
+
+	if pvcNotInUse := v.ensurePVCNotInUse(pvc); !pvcNotInUse {
+		log.Info("Requeueing as PersistentVolumeClaim is still in use")
+
+		return requeue, !skip
+	}
+
+	return !requeue, !skip
+}
+
+// ensurePVCNotInUse ensures PVC is not in use
+func (v *VRGInstance) ensurePVCNotInUse(pvc *corev1.PersistentVolumeClaim) bool {
```

Comment:
Suggestions for alternates to ensurePVCIsSecondary():
- canPVCBeMarkedSecondary()
- shouldPVCBeMarkedSecondary()